### DISCUSSION TITLE
TVHC - Request for help

### DISCUSSION THREAD

**Discussion by SteveDyson** [magiblot/tvision#58] [Q&A] [answered]
Please accept my apologies if this is not the right place for this post.
I am supporting a TV application that I wrote 26 years ago (under DOS). It now runs under Win-XP and with vDos under Win10.
Some of the help information that was written all that time go is now either out of date or downright wrong!
I cannot remember how the help files we written, compiled, generated etc. I have located a load of files like the attached but I cannot remember if these are input for TVHC or outputs! (I have a vague recollection that these are outputs and there should be some text files elsewhere that are inputs, but this is only a vague recollection - 26 years is a long time!)
Does anyone have any info they can forward to me that might help?

As a supplemental question - how stable is the Win-10 release of TV. The application I am supporting is used from 11pm through to 6pm 6 days a week so it is extremely important that whatever I use is extremely stable (I do like my sleep to be uninterrupted :-) ). Should I be considering adopting this port or, at least, evaluating it?

[CUSTSLST.TXT](https://github.com/magiblot/tvision/files/6528087/CUSTSLST.TXT)
[FMAINTNC.TXT](https://github.com/magiblot/tvision/files/6528088/FMAINTNC.TXT)

**Comment by magiblot** [comment, marked as answer]
Hi Steve!

The syntax of these files matches the one used for TVHC inputs. You can find a larger example of a TVHC input in [examples/tvhc/demohelp.txt](https://github.com/magiblot/tvision/blob/master/examples/tvhc/demohelp.txt) (note that special characters are not displayed properly on GitHub because it is unable to figure out the actual file encoding, but they would show properly in the TV application).

However, the TVHC application hosted in this repository accepts a single input file at most. I guess all these files you have would need to be merged before being used as TVHC input.

TVHC generates two files: a `.h` header that is to be used during the compilation of your application, and a `.h16` file with the help information that can be loaded at runtime by the application. Does this sound familiar to you?

Turbo Vision for Windows 10 is probably stable enough to stay working all night long without breaking, but that's not what worries me. Since you mention the application was designed for DOS, the source code will likely depend on operating system features or third-party libraries which are no longer available in modern versions of Windows and thus you won't be able to compile your application with a modern compiler. In that case, you would have to redesign it completely to avoid using these outdated features. Unless this isn't the case, porting this application natively to Windows 10 doesn't seem worth the effort to me.

Another possibility is to rebuild your program as a 32-bit DOS application with the vintage Borland C++ compiler. 32-bit DOS applications do run on Windows 10, so you could get rid of vDos. Borland C++ provides support for several DOS functions in 32-bit mode, so you have more chances that this will work with not many changes in the source code.

So the problem of upgrading your application is less about how reliable it will be and more about how much time will have to be spent to adapt the source code.

Cheers.

**Reply by SteveDyson** [reply]
Thank you Magiblot for such a detailed response. Lots to think about here!

**Comment by WillianBR** [comment]
@SteveDyson  Do you mind sharing your application screens, just for fun of us? I love to look at TV Applications, to see how creative other people are! 

If you don't mind, of course!

**Comment by paule32** [comment]
Hello,
I use notepad++ - it is a free Code-Editor, that can handle many Encodings (ASCII, UTF-8, ...).
Since UniCode support, you can edit/insert special chars, that are 1:1 displayed in the HelpWindow.
notepad++ is a MS-Windows Editor.
I don't know, if it is available under MacOS or other Operating Systems

**Comment by paule32** [comment]
The TV package from magiblot1 is a Windows Application (or better: you can create Windows Applications with TV look&feel).
Take a look to:  https://www.helpndoc.com.

There you can download a Help-Authoring Tool, that can create CHM (Windows Compiled HTML Help "file").
I suggest, you use this CHM file instead of creating TV Help files.
The Tool itself is very easy to use.
You can create Topic's, and Sub-Topics.
You can create Templates, that you can modify/mark-up with the built-in Script Pascal Engine.
For visit/view Help, you can simply start a new process, that loads the CHM file.
The CHM file can be used from Delphi, and/or C++ Builder, and many other, too.
